wetterdienst/provider/noaa/ghcn/api.py

```diff
@@ -0,0 +1,325 @@
+
+        return df_factors
+
+    def _add_derived_parameters(
+        self, df: pd.DataFrame, station_id: str
+    ) -> pd.DataFrame:
+        """
+        Method to add derived parameters to DataFrame
+        :param df: pandas DataFrame with predefined parameters
+        :param parameter: parameter enum to handle how parameter is derived
+        :return:
+        """
+        # TODO: currently only for daily parameters
+        # store parameter names (long accessors)
+        tmax_key = NoaaGhcnParameter.DAILY.TEMPERATURE_AIR_MAX_200.value
+        tmin_key = NoaaGhcnParameter.DAILY.TEMPERATURE_AIR_MIN_200.value
+        tmean_key = NoaaGhcnParameter.DAILY.TEMPERATURE_AIR_MEAN_200.value
+
+        drop_columns = [
+            Columns.STATION_ID.value,
+            Columns.PARAMETER.value,
+            Columns.QUALITY.value,
+        ]
+
+        # TODO: better manage start and end date and its timezone
+        # code taken from ValuesCore._complete_dates
+        timezone_ = self._get_timezone_from_station(station_id)
+
+        # draw tmax and tmin data from DataFrame
+        df_tmax = (
+            df[df[Columns.PARAMETER.value] == tmax_key]
+            .drop(columns=drop_columns)
+            .rename(columns={Columns.VALUE.value: tmax_key})
+        )
+        df_tmin = (
+            df[df[Columns.PARAMETER.value] == tmin_key]
+            .drop(columns=drop_columns)
+            .rename(columns={Columns.VALUE.value: tmin_key})
+        )
+
+        start_date = self.stations.start_date
+        end_date = self.stations.end_date
+
+        if not start_date:
+            # either take start and end date from temperature dfs
+            if not df_tmax.empty or not df_tmin.empty:
+                start_date = np.nanmin(
+                    [
+                        df_tmax[Columns.DATE.value].min(),
+                        df_tmin[Columns.DATE.value].min(),
+                    ]
+                )
+                end_date = np.nanmax(
+                    [
+                        df_tmax[Columns.DATE.value].max(),
+                        df_tmin[Columns.DATE.value].max(),
+                    ]
+                )
+            else:
+                start_date = df[Columns.DATE.value].min()
+                end_date = df[Columns.DATE.value].max()
+
+        tz_delta = self._get_timedelta_from_timezones(timezone_, pytz.UTC)
+
+        df_tmean = pd.DataFrame(
+            {
+                Columns.DATE.value: pd.date_range(
+                    start_date + dt.timedelta(hours=tz_delta),
+                    end_date + dt.timedelta(hours=tz_delta),
+                    freq=self.stations.frequency.value,
+                    tz=pytz.UTC,
+                )
+            }
+        )
+
+        # merge both temperature dfs on dates
+        df_tmean = df_tmean.merge(df_tmax, how="left", on=Columns.DATE.value).merge(
+            df_tmin, how="left", on=Columns.DATE.value
+        )
+
+        df_tmean[Columns.VALUE.value] = (
```

Comment:
Please notice my suggestion above

Reply:
Which one?

Reply:
About how to calculate the mean

Reply:
But what's the point of your suggestion? GHCN doesn't have mean so I take tmax and tmin to calculate the mean.